client/components/site-preview-link/use-site-preview-links.ts

```diff
@@ -23,13 +23,15 @@ export function useSitePreviewLinks( options: UseSitePreviewLinksOptions ) {
 	const { siteId, onSuccess, isEnabled = false } = options;
 	return useQuery< PreviewLinksResponse, unknown, PreviewLink[] >( {
 		queryKey: [ SITE_PREVIEW_LINKS_QUERY_KEY, siteId ],
-		queryFn: () =>
-			wpcom.req.get( {
+		queryFn: async () => {
+			const response = await wpcom.req.get( {
 				path: `/sites/${ siteId }/preview-links`,
 				apiNamespace: 'wpcom/v2',
-			} ),
+			} );
+			onSuccess?.( response );
```

Comment:
That's an interesting solution I haven't thought about. Adding a side effect here might cause troubles though and is not something we currently recommended. It's better to wrap those in `useEffect` handlers (especially when you're calling `setState` in this callback). To know whether a query is in `success` state you can compare `status === success` or use `isSuccess` returned from the query.

Reply:
Thanks for the feedback. I replaced the `onSuccess` with the `useEffect` https://github.com/Automattic/wp-calypso/pull/84658/commits/fd6b069523ddf9d1358a09be7c35a2afd8b3ead3